File: ExampleApp/app/src/main/java/com/motiv/example/PostDetailsActivityPresenter.java
package com.motiv.example;

import com.motiv.example.dao.DaoRepository;
import com.motiv.example.dao.LocalStorage;

public class PostDetailsActivityPresenter implements PostDetailsActivityContract.Presenter {

    private PostDetailsActivityContract.View view;
    private GoApi goApi;
    private AuthApi authApi;
    private DaoRepository daoRepository;
    private LocalStorage localStorage;

    public PostDetailsActivityPresenter(
            PostDetailsActivityContract.View view,
            GoApi goApi,
            AuthApi authApi,
            DaoRepository daoRepository,
            LocalStorage localStorage) {
        this.view = view;
        this.goApi = goApi;
        this.authApi = authApi;
        this.daoRepository = daoRepository;
        this.localStorage = localStorage;
    }

    @Override
    public void daoRepositoryloadPost(java.lang.String arg0) {
        daoRepository.loadPost(
                arg0,
                new com.motiv.example.OnResponseListener<com.motiv.example.Post>() {
                    @Override
                    public void onSuccess(com.motiv.example.Post argument0) {
                        view.textview10setText(argument0.getTitle());
                        view.textview11setText(argument0.getBody());
                    }

                    @Override
                    public void onError(Exception argument0) {}
                });
    }
}

File: ExampleApp/app/src/main/java/com/motiv/example/PostDetailsActivityPresenter.kt
 
package com.motiv.example
import androidx.fragment.app.*
import com.motiv.example.dao.DaoRepository
import com.motiv.example.dao.LocalStorage
import dagger.*
import dagger.android.*
import dagger.android.support.*
import javax.inject.*

public class PostDetailsActivityPresenter(val view: PostDetailsActivityContract.View, val goApi: GoApi, val authApi: AuthApi, val daoRepository: DaoRepository, val localStorage: LocalStorage) : PostDetailsActivityContract.Presenter {

    override fun daoRepositoryloadPost(arg0: String) {
        daoRepository.loadPost(
            arg0,
            object : com.motiv.example.OnResponseListener<com.motiv.example.Post> {
                override fun onSuccess(argument0: com.motiv.example.Post) {
                    view.textview10setText(argument0.getTitle())
                    view.textview11setText(argument0.getBody())
                } override fun onError(argument0: Exception) {
                } 
            }
        )
    }
}
